### DISCUSSION TITLE
Integration to an existing Laravel application. Call to a member function handle() on null

### DISCUSSION THREAD

**Discussion by GuyVervillePerso** [statamic/cms#8299] [Troubleshooting]
I carefully followed the instructions for installing Statamic in a Laravel application. Everything seems to work fine (after modifying the migration offered by Statamic, which doesn't take into account that the "users" table already exists.) 

If I try to connect with the already set login page (not the Statamic one) or if I try to go to the Control Panel section of Statamic trying to log to (/cp), I get the error _Call to a member function handle() on null_ without any further explanation.

I really don't know what to look for...

Note that I'm correctly connected with the Laravel login, but that the error I get on the destination page is with the @can... command (see image). But if I go to /cp, I still get an error with no further details. Should I try to resolve the Policies first?

<img width="904" alt="image" src="https://github.com/statamic/cms/assets/69694028/22f7e3c7-1999-4065-a564-a7fdb88930a7">

N.B. I wonder if those permissions are simply not read by Statamic even it is supposed to read the BD table...

The code for the Gates resides in a Middleware

`class AuthGates
{
    public function handle($request, Closure $next)
    {
        $user = auth()->user();

        if (!$user) {
            return $next($request);
        }

        $roles            = Role::with('permissions')->get();
        $permissionsArray = [];

        foreach ($roles as $role) {
            foreach ($role->permissions as $permissions) {
                $permissionsArray[$permissions->title][] = $role->id;
            }
        }
        foreach ($permissionsArray as $title => $roles) {
            Gate::define($title, function ($user) use ($roles) {
                return count(array_intersect($user->roles->pluck('id')->toArray(), $roles)) > 0;
            });
        }
        return $next($request);
    }
}
`
But more importantly, I do not have access at the /cp even if I remove temporarily the @can command for the main page. The Statamic Control Panel is independant of my application, I presume, and I get the same error...

https://flareapp.io/share/Lm8zGRk7

**Comment by ryanmitchell** [comment]
It's likely to be that it cant find the group associated with the user, which is a bit of known issue. I have an open PR that tries to solve this: https://github.com/statamic/cms/pull/5686 

Maybe you could trying pulling it in and seeing if it fits your needs?

**Comment by GuyVervillePerso** [comment]
Thank you for answering. There is hope, I can see. I am not familiar to operate PR from an external repo (since I have my own repo for that project). If you were kind enough to point me to some instructions, I would appreciate it.

**Reply by ryanmitchell** [reply]
Theres a good blog post on composer patches which may help you out: https://rias.be/blog/using-composer-patches

**Comment by GuyVervillePerso** [comment]
Thank you!

**Comment by GuyVervillePerso** [comment]
I tried the solution you provided with the help of the diff packaged you mentioned. I realize that your pr is for Statamic 3.3. I am with Statamic 4. It seems not to be applied. :-(
